### PR TITLE
consistent rq status naming and handling

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
## What type of PR is this? 
- support more RQ statuses (STOPPED, CANCELED, DEFERRED, etc)
- consistent status naming
- replaced deprecated job result handling methods

- [x] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] New Query Runner (Data Source) 
- [ ] New Alert Destination
- [ ] Other

## How is this tested?

- [ ] Unit tests (pytest, jest)
- [ ] E2E Tests (Cypress)
- [ ] Manually
- [ ] N/A

### CODE REVIEW

**Comment by justinclift** [comment]
Looks like it needs a bit more attention in order to pass CI. :wink:

**Comment by AndrewChubatiuk** [comment]
> Looks like it needs a bit more attention in order to pass CI. 😉

just had to update a test

**Comment by justinclift** [comment]
@gaecoli @wlach Eyeballing this, it seems sensible.  You're both better at Python than me though, so would either of you be ok to take a quick look over this to make sure there's nothing I missed? :smile:

File: redash/serializers/__init__.py [hunk 1]
```diff
@@ -271,46 +272,26 @@ def serialize(self):
...
+        else:
+            error = job_result.exc_string
 
     return {
         "job": {
             "id": job.id,
             "updated_at": updated_at,
             "status": status,
             "error": error,
-            "result": result,
```
[the remark this answers is not in the document]
Comment (gaecoli) [reply redash/serializers/__init__.py:313]: Will it be affected if the user gets the results directly through the API?🤔

[the remark this answers is not in the document]
Comment (gaecoli) [reply redash/serializers/__init__.py:313]: @AndrewChubatiuk I'm a little confused, thank you for helping me out.

[the remark this answers is not in the document]
Comment (AndrewChubatiuk) [reply redash/serializers/__init__.py:313]: result and query_result_id returned the same value - id in query_results table. to have no confusion I removed redundancy and renamed query_result_id to just result_id

[the remark this answers is not in the document]
Comment (gaecoli) [reply redash/serializers/__init__.py:313]: > result and query_result_id returned the same value - id in query_results table. to have no confusion I removed redundancy and renamed query_result_id to just result_id

Ok, i have no problem.

[the remark this answers is not in the document]
Comment (AndrewChubatiuk) [reply redash/serializers/__init__.py:313]: @gaecoli could you please review?

**Review by gaecoli — APPROVED** [review]
LGTM!

**Pull request merged**